### DISCUSSION TITLE
Execute API call from Azure DevOps agent.

### DISCUSSION THREAD

**Discussion by stas-sultanov** [microsoft/powerplatform-build-tools#1047] [Q&A]
Greetings.
I am trying to automate provision of Power Platform environment.
As part of automation I would like to register an Azure Managed Identity within the Power Platform tenant. According to documentation this feature is in preview and can be achieved only via API call.

The questions are:
 1. how I can execute a custom API call via pac?
 2. Can I get a bearer token to make the request with help of powershell?
 3. How I can do this on AzureDevOps agent/pipeline
thanks.

**Comment by petrochuk** [comment]
1. PAC CLI doesn't support calling arbitrary APIs. You can try using Dataverse Service Client or Invoke-WebRequest
2. This is good page with different ways to [acquire an access token](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token) and here is section for [PowerShell](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-powershell)
3. It is possible to add PowerShell step with your custom script to Azure DevOps pipeline after that

**Comment by stas-sultanov** [comment]
@petrochuk, thanks for information.

We can get a Bearer token for Service Connection of Azure Resource Manager type
with help of [AzurePowerShell](https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/azure-powershell-v5) task and [Get-AzAccessToken](https://learn.microsoft.com/en-us/powershell/module/az.accounts/get-azaccesstoken?view=azps-12.3.0) command.

The problem is that Power Tools tasks use Service Connection of Power Platform type, not Azure Resource Manager type.

My understanding is that either Power Tools should use Service Connection of Azure Resource Manager type or there should be a PowerPlatformPowerShell task that supports Service Connection of Power Platform type.
